src/NewButton/icon-button.tsx

```diff
@@ -0,0 +1,31 @@
+import React, {forwardRef} from 'react'
+import {merge, SxProp} from '../sx'
+import {useTheme} from '../ThemeProvider'
+import Box from '../Box'
+import {IconButtonProps} from './types'
+import {getBaseStyles, getSizeStyles, getVariantStyles} from './styles'
+import {Base} from './button-base'
+
+const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>((props, forwardedRef): JSX.Element => {
+  const {variant = 'default', size = 'medium', sx: sxProp = {}, icon: Icon, iconLabel} = props
+  const {theme} = useTheme()
+  const styles = {
+    ...getBaseStyles(theme)
+  }
+  const sxStyles = merge.all([
+    styles,
+    getSizeStyles(size, variant, true),
+    getVariantStyles(variant, theme),
+    sxProp as SxProp
+  ])
+  return (
+    <Base sx={sxStyles} ref={forwardedRef} {...props}>
+      <span hidden={true}>{iconLabel}</span>
```

Comment:
Is `hidden` needed anymore? If so, should it be `aria-hidden` instead?

Reply:
Since its an icon only component, the idea was to have a visibly hidden icon label but the screenreaders will be able to access it.

Reply:
Okay, I believe the `hidden` html5 attribute will also hide it from screenreaders too though, because [it prevents the semantics from getting surfaced to the accessibility api  ](https://www.w3.org/TR/core-aam/#exclude_elements2). I think you'll either need to just visually hide it with styles, or explicitly add `aria-hidden="false"`, but the latter seems counter-productive.